### ISSUE TITLE
Sandbox example errors: React - TanStack Router (App) - SSR

### ISSUE THREAD

**Issue by redbar0n** [nksaraf/vinxi#111]
Whereas the React SSR Basic sandbox example works fine, this is what I get for [the SSR version](https://camo.githubusercontent.com/7f5709ecf7bd8016ce151719ee2ce57774399bd74ec3288c1a393ed1d951c8e4/68747470733a2f2f646576656c6f7065722e737461636b626c69747a2e636f6d2f696d672f6f70656e5f696e5f737461636b626c69747a2e737667):

> ~/projects/kkvipxjky.github
> ❯ npm install && npm run dev
> npm WARN ERESOLVE overriding peer dependency
> npm WARN While resolving: @gisatcz/cross-package-react-context@0.2.0
> npm WARN Found: react@0.0.0-experimental-035a41c4e-20230704
> npm WARN node_modules/react
> npm WARN   react@"0.0.0-experimental-035a41c4e-20230704" from the root project
> npm WARN   1 more (react-dom)
> npm WARN 
> npm WARN Could not resolve dependency:
> npm WARN peer react@"^16.13.1 || ^17.0.2 || ^18.1.0" from @gisatcz/cross-package-react-context@0.2.0
> npm WARN node_modules/@gisatcz/cross-package-react-context
> npm WARN   @gisatcz/cross-package-react-context@"^0.2.0" from @tanstack/react-start@0.0.1-beta.111
> npm WARN   node_modules/@tanstack/react-start
> npm WARN   1 more (@tanstack/router)
> npm WARN 
> npm WARN Conflicting peer dependency: react@18.2.0
> npm WARN node_modules/react
> npm WARN   peer react@"^16.13.1 || ^17.0.2 || ^18.1.0" from @gisatcz/cross-package-react-context@0.2.0
> npm WARN   node_modules/@gisatcz/cross-package-react-context
> npm WARN     @gisatcz/cross-package-react-context@"^0.2.0" from @tanstack/react-start@0.0.1-beta.111
> npm WARN     node_modules/@tanstack/react-start
> npm WARN     1 more (@tanstack/router)
> npm ERR! code ERESOLVE
> npm ERR! ERESOLVE could not resolve
> npm ERR! 
> npm ERR! While resolving: @tanstack/react-loaders@0.0.1-beta.109
> npm ERR! Found: react@0.0.0-experimental-035a41c4e-20230704
> npm ERR! node_modules/react
> npm ERR!   react@"0.0.0-experimental-035a41c4e-20230704" from the root project
> npm ERR!   peer react@"0.0.0-experimental-035a41c4e-20230704" from react-dom@0.0.0-experimental-035a41c4e-20230704
> npm ERR!   node_modules/react-dom
> npm ERR!     react-dom@"0.0.0-experimental-035a41c4e-20230704" from the root project
> npm ERR! 
> npm ERR! Could not resolve dependency:
> npm ERR! peer react@">=16" from @tanstack/react-loaders@0.0.1-beta.109
> npm ERR! node_modules/@tanstack/react-loaders
> npm ERR!   @tanstack/react-loaders@"0.0.1-beta.109" from the root project
> npm ERR!   @tanstack/react-loaders@"0.0.1-beta.109" from @tanstack/react-start@0.0.1-beta.111
> npm ERR!   node_modules/@tanstack/react-start
> npm ERR!     @tanstack/react-start@"0.0.1-beta.111" from the root project
> npm ERR! 
> npm ERR! Conflicting peer dependency: react@18.2.0
> npm ERR! node_modules/react
> npm ERR!   peer react@">=16" from @tanstack/react-loaders@0.0.1-beta.109
> npm ERR!   node_modules/@tanstack/react-loaders
> npm ERR!     @tanstack/react-loaders@"0.0.1-beta.109" from the root project
> npm ERR!     @tanstack/react-loaders@"0.0.1-beta.109" from @tanstack/react-start@0.0.1-beta.111
> npm ERR!     node_modules/@tanstack/react-start
> npm ERR!       @tanstack/react-start@"0.0.1-beta.111" from the root project
> npm ERR! 
> npm ERR! Fix the upstream dependency conflict, or retry
> npm ERR! this command with --force or --legacy-peer-deps
> npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
> npm ERR! 
> npm ERR! 
> npm ERR! For a full report see:
> npm ERR! /home/.npm/_logs/2024-01-10T14_33_28_197Z-eresolve-report.txt
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/.npm/_logs/2024-01-10T14_33_28_197Z-debug-0.log
> 
> ~/projects/kkvipxjky.github 2s
> ❯ npm install --legacy-peer-deps && npm run dev
> 
> added 650 packages in 6s
> 
> 123 packages are looking for funding
>   run `npm fund` for details
> 
> > dev
> > vinxi dev
> 
> vinxi 0.1.1
> vinxi starting dev server

and then nothing displays in the browser view. This is on macOS with Firefox 115.6.0esr (64-bit).

**Comment by redbar0n** [comment]
Might be because it's on an ancient version of TanStack Router:

package.json:

>   "@tanstack/router": "0.0.1-beta.111",

whereas TanStack Router is now at v1.3.0.

**Comment by nksaraf** [comment]
Yeah the tanstack router needs updating. I wonder if we can use their new vite plugin.

**Comment by redbar0n** [comment]
Yeah it was released yesterday: https://x.com/tannerlinsley/status/1744496785815662751?s=46&t=tYj6dxrnDA8GvBz3qad13A

Tanner is looking for feature requests for it, so you might have success by suggesting Vinxi integration with him: https://x.com/tannerlinsley/status/1743840174319243763?s=46&t=tYj6dxrnDA8GvBz3qad13A

PS: Personally, I got a bit more excited about SSRx than Vinxi because it is server agnostic, so I could use Hono/Hattip or some (other) super fast Node/Bun server with it if I want.

**Comment by nksaraf** [comment]
While I can understand the urge to go with different server runtimes. I really think a stable standard server runtime is important for vinxis plan to be versatile across platforms and frameworks. H3 is very optimised on its own and powers nuxt one of the faster servers. It uses a very very thin layer on the bare platform server (be it node or bun). So in a lot of cases it's faster than other node servers that adapt to the web fetch API.

**Comment by redbar0n** [comment]
In what way is a standard server runtime necessary to be versatile across platforms and frameworks? Both SSRx and Vike are just as versatile wrt. platforms/deploy-targets, no? And in the worst case then frameworks already integrate with their own server runtime (like Nitro does with H3..).

Have a look at Hono vs H3 in [benchmarks](https://github.com/SaltyAom/bun-http-framework-benchmark), where Hono outperforms H3 (not on Node though). Likely since [H3 doesn't run on Bun yet](https://github.com/unjs/h3/issues/425) (?). Hono uses a [faster](https://hono.dev/concepts/routers) [RegExpRouter](https://hono.dev/concepts/routers#regexprouter) instead of a [Radix Tree](https://en.wikipedia.org/wiki/Radix_tree) based router such as [Radix3](https://github.com/unjs/radix3) which H3 uses. (You can similarly use [TrieRouter](https://hono.dev/concepts/routers#trierouter) in Hono if you like).

**Comment by nksaraf** [comment]
> where Hono outperforms H3 (not on Node though)

Node's important though. Probably still the most widely used.


> Likely since [H3 doesn't run on Bun yet](https://github.com/unjs/h3/issues/425) (?)

It does. You can run the whole vinxi stack natively in bun using `bun run --bun dev`, `bun run --bun build` and `bun run --bun start`.

> In what way is a standard server runtime necessary to be versatile across platforms and frameworks?

It allows us to build a shared library of trusted tools that can do the common operations across all the platforms, so sessions, cors, cookies, auth, etc. It prevents us from having to open too many doors everywhere and make a lot of assumptions about the server code you are writing. This allows us to keep vinxi much simpler (yet really powerful) than it would need to be to support any arbitrary server runtime.

From vinxi's perspective, it's way better to bet on `h3` which is backed by the `nitro` and `unjs` teams and is used by `analog` and `nuxt`. So lot of people working on keep it stable, fast, bug free and working everywhere. (and its still quite a difficult task).

**Comment by nksaraf** [comment]
Also if TrieRouter's really good we should try to push an implementation upstream to `radix3` or not use `radix3` and just implement the routing using trierouter in a h3 handler. `h3` is just my interface to the rest of the HTTP world from my server.
